src/integration/java/io/pinecone/integration/dataPlane/UpdateFetchAndQueryPodTest.java

```diff
@@ -121,49 +115,50 @@ public void updateAllParamsFetchAndQuerySyncTest() throws InterruptedException {
                     null,
                     true,
                     true);
+            List<ScoredVectorWithUnsignedIndices> queryResults = queryResponse.getMatchesList();
+            ScoredVectorWithUnsignedIndices scoredVectorV1 = null;
+
+            for (ScoredVectorWithUnsignedIndices vector : queryResults) {
+                if (idToUpdate.equals(vector.getId())) {
+                    scoredVectorV1 = vector;
+                }
+            }
 
-            ScoredVectorWithUnsignedIndices scoredVectorV1 = queryResponse.getMatches(0);
             // Verify the correct vector id was updated
+            assertNotNull(scoredVectorV1);
             assertEquals(scoredVectorV1.getId(), idToUpdate);
 
             // Verify the updated values
-            List<Float> valuesList = new ArrayList<>(scoredVectorV1.getValuesList());
-            Collections.sort(valuesList);
-            Collections.sort(valuesToUpdate);
+            List<Float> valuesList = scoredVectorV1.getValuesList();
```

Comment:
I don't know why these were being sorted, and a lot of this was just really confusing to me so I tried to simplify things.

Reply:
awesomeness